### PR TITLE
Cav 574

### PR DESCRIPTION
**Pull request by mar36tin**

### CODE REVIEW

File: app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+package uk.gov.hmrc.cipemail.controllers
+
+import uk.gov.hmrc.internalauth.client.{IAAction, Predicate, Resource, ResourceLocation, ResourceType}
+
+object InternalAuthAccess {
+  val permission: Predicate.Permission = Predicate.Permission(Resource(
+    ResourceType("cip-email"),
+    ResourceLocation("")),
```
Comment (johno1985) [inline app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala:24]: should be *

File: app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    IAAction("READ"))
```
Comment (johno1985) [inline app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala:25]: should be *

**Review by johno1985 — CHANGES_REQUESTED** [review]
minor comments and changes required

File: app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala (hunk 2 above)
Comment (mar36tin) [reply app/uk/gov/hmrc/cipemail/controllers/InternalAuthAccess.scala]: done

File: test/uk/gov/hmrc/cipemail/controllers/NotificationControllerSpec.scala [hunk 3]
```diff
@@ -61,20 +60,34 @@ class NotificationControllerSpec extends AnyWordSpec
...
       val response = controller.status("test-notification-id")(fakeRequest)
       mockMetricsService.recordMetric("cip-notification-status-failure") was called
       status(response) shouldBe GATEWAY_TIMEOUT
     }
   }
+
+  trait SetUp {
+    protected val fakeRequest = FakeRequest().withHeaders("Authorization" -> "fake-token")
+    val expectedPredicate = {
+      Permission(Resource(ResourceType("cip-email"), ResourceLocation("")), IAAction("READ"))
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/controllers/NotificationControllerSpec.scala:84]: needs updating else test will fail

File: test/uk/gov/hmrc/cipemail/controllers/VerifyControllerSpec.scala [hunk 4]
```diff
@@ -98,5 +95,20 @@ class VerifyControllerSpec extends AnyWordSpec
       status(response) shouldBe GATEWAY_TIMEOUT
     }
   }
+
+  trait SetUp {
+    val headerName = "header-name"
+    val headerValue = "header-value"
+    protected val fakeRequest = FakeRequest().withHeaders("Authorization" -> "fake-token")
+    val expectedPredicate = {
+      Permission(Resource(ResourceType("cip-email"), ResourceLocation("")), IAAction("READ"))
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/controllers/VerifyControllerSpec.scala:104]: needs updating else test will fail

File: test/uk/gov/hmrc/cipemail/controllers/VerifyPasscodeControllerSpec.scala [hunk 5]
```diff
@@ -91,4 +90,17 @@ class VerifyPasscodeControllerSpec extends AnyWordSpec
       status(response) shouldBe GATEWAY_TIMEOUT
     }
   }
+
+  trait SetUp {
+    protected val fakeRequest = FakeRequest().withHeaders("Authorization" -> "fake-token")
+    val expectedPredicate = {
+      Permission(Resource(ResourceType("cip-email"), ResourceLocation("")), IAAction("READ"))
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipemail/controllers/VerifyPasscodeControllerSpec.scala:97]: needs updating else test will fail

**Comment by johno1985** [comment]
still one last failing unit test, to do with metrics mock

**Review by johno1985 — APPROVED** [review]

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
unit test failures

**Comment by johno1985** [comment]
rebuild

**Review by johno1985 — APPROVED** [review]

**Pull request merged**